rclpy/rclpy/node.py

```diff
@@ -113,11 +113,12 @@ def __init__(
             args.
         :param start_parameter_services: ``False`` if the node should not create parameter
             services.
-        :param initial_parameters: A list of parameters to be set during node creation.
+        :param parameter_overrides: A list of overrides for initial values for parameters declared
```

Comment:
Same comment as on `rclcpp`, I think that mentioning "initial values" is misleading, because these behave more like the final values.

Reply:
They are not final values... They only apply when being declared. If you declare it and then set it afterwards the override doesn't affect the set.

Reply:
Here's an example I made just to make sure I was right in my thinking:

```python
import rclpy

rclpy.init()

n1 = rclpy.create_node('test')
print("Parameter 'foo' after declare:", n1.declare_parameter('foo', 'bar').value)
n1.set_parameters([rclpy.Parameter('foo', rclpy.Parameter.Type.STRING, 'ping')])
print("Parameter 'foo' after set:", n1.get_parameter('foo').value)

n2 = rclpy.create_node(
    'test2',
    automatically_declare_parameters_from_overrides=False,
    parameter_overrides={rclpy.Parameter('foo', rclpy.Parameter.Type.STRING, 'baz')})
print("Parameter 'foo' after declare:", n2.declare_parameter('foo', 'bar').value)
n2.set_parameters([rclpy.Parameter('foo', rclpy.Parameter.Type.STRING, 'ping')])
print("Parameter 'foo' after set:", n2.get_parameter('foo').value)
```

Which produces:

```
Parameter 'foo' after declare: bar
Parameter 'foo' after set: ping
Parameter 'foo' after declare: baz
Parameter 'foo' after set: ping
```

----

@jubeira after writing this little toy program to check the behavior @chapulina and I were talking about, I realized the parameter API in rclpy is not that friendly to use...

I'll make a new issue to track this, but I think we'll have some follow up work to do after Dashing on that part of the interface.

Reply:
See https://github.com/ros2/rclcpp/pull/745#discussion_r288344793 and the `foo == baz` behaviour on https://github.com/ros2/rclcpp/issues/730#issuecomment-495801806

Reply:
I think it's a false alarm, see my response: https://github.com/ros2/rclcpp/pull/745#discussion_r288353024

Reply:
> @jubeira after writing this little toy program to check the behavior @chapulina and I were talking about, I realized the parameter API in rclpy is not that friendly to use...
> 
> I'll make a new issue to track this, but I think we'll have some follow up work to do after Dashing on that part of the interface.

Sounds good.
From that example, I think there are two things that contribute to that feeling:
- Having to specify the `Parameter.Type` when creating a `Parameter` object, instead of guessing it automatically according to the given value. This was addressed in #347 without breaking the existing API:
```python
    # These two are equivalent:
    rclpy.Parameter('foo', rclpy.Parameter.Type.STRING, 'baz')
    rclpy.Parameter('foo', value='baz')
```
While I think this looks better than before, perhaps there's still room for improvement.
- There is an asymmetry between `declare` and `set` methods: the former ones use a tuple with name, value (and descriptor), and the latter ones use `rclpy.Parameter` as arguments. Following that line, maybe a version of `declare_parameter(s)` accepting a `rclpy.Parameter` and a descriptor would work (as indicated in https://github.com/ros2/rclpy/pull/325#discussion_r280534305). The problem with that approach was that `declare_parameters` accepts a namespace, and the code that actually sets the parameter downstream takes a `Parameter` is not prepared to handle it so it would not be a trivial change.

Let me know if you want to discuss this in a brief meeting; I'd like to hear your opinion about this.